scripts/offline_data_processing.py

```diff
@@ -0,0 +1,257 @@
+):
+    """Process dataset based on config yaml
+
+    Args:
+        model_args: tuning.config.configs.ModelArguments
+        data_args: tuning.config.configs.DataArguments
+        train_args: tuning.config.configs.TrainingArguments
+    """
+    # Set log level for this function
+    train_args, logger = set_log_level(train_args, "get_processed_dataset")
+
+    logger.info(
+        "Starting dataset processing with model_args: %s, data_args: %s, training_args: %s",
+        model_args,
+        data_args,
+        train_args,
+    )
+
+    # Load tokenizer for the model
+    tokenizer_path = (
+        model_args.tokenizer_name_or_path
+        if model_args.tokenizer_name_or_path
+        else model_args.model_name_or_path
+    )
+    logger.debug("Loading tokenizer from %s", tokenizer_path)
+    tokenizer = AutoTokenizer.from_pretrained(
+        tokenizer_path,
+        cache_dir=train_args.cache_dir,
+        use_fast=True,
+        legacy=True,
+    )
+    logger.debug("Tokenizer loaded successfully.")
+
+    # Add chat_template to the tokenizer
+    if data_args.chat_template:
+        logger.info("Adding chat_template to the tokenizer.")
+        if tokenizer.chat_template:
+            logger.warning(
+                "Replacing existing chat_template %s with the given chat_template %s",
+                tokenizer.chat_template,
+                data_args.chat_template,
+            )
+        tokenizer.chat_template = data_args.chat_template
+
+    # Prepare special tokens dictionary
+    special_tokens_dict = {}
+    if not model_args.tokenizer_name_or_path:
+        if isinstance(tokenizer, (LlamaTokenizer, LlamaTokenizerFast)):
+            logger.debug(
+                "Using a Llama tokenizer—setting eos_token to </s> by default."
+            )
+            special_tokens_dict["eos_token"] = "</s>"
+
+    if not model_args.tokenizer_name_or_path:
+        if tokenizer.pad_token is None:
+            logger.warning(
+                "PAD token not found in tokenizer; setting PAD token to default."
+            )
+            special_tokens_dict["pad_token"] = configs.DEFAULT_PAD_TOKEN
+        if tokenizer.eos_token is None:
+            logger.warning(
+                "EOS token not found in tokenizer; setting EOS token to default."
+            )
+            special_tokens_dict["eos_token"] = configs.DEFAULT_EOS_TOKEN
+        if tokenizer.pad_token == tokenizer.eos_token:
+            logger.warning(
+                "PAD token and EOS token are currently the same. Overriding PAD token or EOS token."
+            )
+            if tokenizer.eos_token != configs.DEFAULT_PAD_TOKEN:
+                tokenizer.pad_token = configs.DEFAULT_PAD_TOKEN
+                special_tokens_dict["pad_token"] = configs.DEFAULT_PAD_TOKEN
+            else:
+                tokenizer.eos_token = configs.DEFAULT_EOS_TOKEN
+                special_tokens_dict["eos_token"] = configs.DEFAULT_EOS_TOKEN
+
+    if special_tokens_dict:
```

Comment:
you will have to expand this to include the special tokens via command line support being added by #470